packages/slate/src/models/node.js

```diff
@@ -1982,14 +2017,34 @@ class Node {
    * @return {Node}
    */
 
-  updateNode(node) {
+  updateNode(node, willTreeChanged = true, changedChildIndex = 0) {
```

Comment:
I'm not really keen on these extra arguments that are hard to remember, and very hard to use if you aren't deeply familiar with the code. Is there any way to do this using the existing recursive, top-down memoization techniques?